libnmstate/state.py

```diff
@@ -80,6 +107,28 @@ def interfaces(self):
         """ Indexed interfaces state """
         return self._ifaces_state
 
+    @property
+    def routes(self):
+        """
+        The list of route in config. Can read and write.
+        """
+        return self._routes
+
+    @routes.setter
+    def routes(self, value):
+        """
+        Replace the routes
+        """
+        self._routes = value
+
+    @property
+    def iface_routes(self):
+        """
+        Indexed routes by next hop interface name. Read only.
+        """
+        self._routes.sort(key=_route_sort_key)
+        return State._index_routes_by_iface(self._routes)
```

Comment:
Why there is a need to re-index the routes each time we access this property?
It is also unclear why the sorting is done as part of the indexing.

I guess the original problem here is that this function does two things instead of one.

Reply:
I changed the code to only do sorting when at verification stage.